pkg/apis/core/v1/helper/helpers.go

```diff
@@ -315,34 +316,45 @@ func NodeSelectorRequirementKeysExistInNodeSelectorTerms(reqs []v1.NodeSelectorR
 // MatchNodeSelectorTerms checks whether the node labels and fields match node selector terms in ORed;
 // nil or empty term matches no objects.
 func MatchNodeSelectorTerms(
+	node *v1.Node,
 	nodeSelectorTerms []v1.NodeSelectorTerm,
```

Comment:
Actually, could you make it `v1.NodeSelector`? https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2610:6